src/WebJobs.Script/Rpc/ILanguageWorkerChannelManager.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Script.Diagnostics;
+
+namespace Microsoft.Azure.WebJobs.Script.Rpc
+{
+    public interface ILanguageWorkerChannelManager
+    {
+        Task InitializeWebhostChannelsAsync(IEnumerable<string> languages);
+
+        Task InitializeWebhostChannelAsync(string language);
+
+        ILanguageWorkerChannel GetWebHostChannel(string language);
+
+        Task ReloadEnvironmentVariablesAsync(string language);
+
+        bool ShutdownStandbyChannelIfExists(string language);
+
+        Task ShutdownStandhbyChannelsAsync();
+
+        ILanguageWorkerChannel CreateLanguageWorkerChannel(string workerId, string scriptRootPath, string language, bool isWebHostChannel, int attemptCount);
```

Comment:
How do these Create methods relate to the Get method above? Will Get fail if Create hasn't been called previously? Can there just be a Get method that internally does the create if not exists logic?

Reply:
- CreateLanguageWorkerChannel --> Creates a new language worker channel but **does not start worker process**
- InitializeChannelAsync --> does the following
  - Create language worker channel
  - Starts worker process
  -  waits for the worker process to send first "Start Stream" message. This is the first message that establishes the bidirectional grpc channel between host and a language worker